android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java

```diff
@@ -303,6 +306,59 @@ public TiListView(TiViewProxy proxy, Activity activity) {
 		listView.setFocusableInTouchMode(true);
 		listView.setDescendantFocusability(ViewGroup.FOCUS_AFTER_DESCENDANTS);
 
+		final TiViewProxy fProxy = proxy;
+		listView.setOnScrollListener(new OnScrollListener()
+		{
+			private int _firstVisibleItem = 0;
+			private int _visibleItemCount = 0;
+			private boolean canFireScrollStart = true;
+			private boolean canFireScrollEnd = false;
+
+			@Override
+			public void onScrollStateChanged(AbsListView view, int scrollState)
+			{
+				String eventName;
+				if (scrollState == OnScrollListener.SCROLL_STATE_IDLE && canFireScrollEnd) {
+					eventName = TiC.EVENT_SCROLLEND;
+					canFireScrollEnd = false;
+					canFireScrollStart = true;
+				} else if (scrollState == OnScrollListener.SCROLL_STATE_TOUCH_SCROLL && canFireScrollStart) {
+					eventName = TiC.EVENT_SCROLLSTART;					
+					canFireScrollEnd = true;
+					canFireScrollStart = false;
+				} else {
+					return;
+				}
+
+				KrollDict eventArgs = new KrollDict();
+				Pair<ListSectionProxy, Pair<Integer, Integer>> info = getSectionInfoByEntryIndex(_firstVisibleItem);
+				int visibleItemCount = _visibleItemCount;
+				
+				int itemIndex = info.second.second;
+				ListSectionProxy section = info.first;
+				
+				if (section.getHeaderTitle() == null || section.getHeaderView() == null) {
```

Comment:
getSectionInfoByEntryIndex should take header and footer into account already. This will cause index to be 1 off when header is present